### PR TITLE
Added iDeferLoading option

### PR DESCRIPTION
**Pull request by nicovogelaar**

### CODE REVIEW

File: src/Pike/DataTable/Adapter/DataTables.php [hunk 1]
```diff
@@ -33,10 +38,21 @@ public function __construct()
      */
     public function render(DataTable $dataTable)
     {
+        $optionsEncoded = Json\Json::encode($this->options, false, array(
+                    'enableJsonExprFinder' => true
+                ));
```
Comment (keesschepers) [inline src/Pike/DataTable/Adapter/DataTables.php]: Shouldn't the parentheses be alignt 4 spaces to the left here?

File: src/Pike/DataTable/Adapter/DataTables.php [hunk 2]
```diff
@@ -33,10 +38,21 @@ public function __construct()
...
+
         $this->viewModel->setVariable('id', $this->id);
         $this->viewModel->setVariable('options', $this->options);
+        $this->viewModel->setVariable('optionsEncoded', $optionsEncoded);
         $this->viewModel->setVariable('columns', $this->getColumnBag());
 
+        if ($this->getOption('iDeferLoading')) {
```
Comment (keesschepers) [inline src/Pike/DataTable/Adapter/DataTables.php]: What does getOption() return? I think I like the if (null !== $this->getOption()) more!

File: src/Pike/DataTable/Adapter/DataTables.php (hunk 1 above)
Comment (nicovogelaar) [inline src/Pike/DataTable/Adapter/DataTables.php]: Netbeans Auto-Indentation does this

File: src/Pike/DataTable/Adapter/DataTables.php (hunk 2 above)
Comment (nicovogelaar) [inline src/Pike/DataTable/Adapter/DataTables.php]: Yes, that's better :)

File: src/Pike/DataTable/Adapter/DataTables.php (hunk 1 above)
Comment (keesschepers) [inline src/Pike/DataTable/Adapter/DataTables.php]: Still I would like to see the indention PSR-2 compatible ;) Why don't you run this program over it? https://github.com/fabpot/PHP-CS-Fixer

**Comment by pietervogelaar** [comment]
Do you have any estimate when the feedback will be fixed? I really like to see this merged!

File: src/Pike/DataTable/Adapter/AbstractAdapter.php [hunk 3]
```diff
@@ -173,7 +173,7 @@ public function getFilter($name)
     /**
      * Sets the filter for auto escaping column strings
      *
-     * @param  integer $priority
+     * @param  integer          $priority
```
Comment (pietervogelaar) [inline src/Pike/DataTable/Adapter/AbstractAdapter.php:176]: Return altijd op een nieuwe regel laten beginnen. Dan krijg je ook deze uitlijning niet.

File: src/Pike/DataTable/DataSource/Doctrine/WhereLikeWalker.php [hunk 4]
```diff
@@ -142,56 +142,56 @@ protected function _getConditionalExpression(PathExpression $pathExpression,
 //                case 'eq':
```
Comment (keesschepers) [inline src/Pike/DataTable/DataSource/Doctrine/WhereLikeWalker.php:142]: All this code is commented. Shouldn't we remove it?

File: src/Pike/DataTable/Adapter/AbstractAdapter.php (hunk 3 above)
Comment (keesschepers) [inline src/Pike/DataTable/Adapter/AbstractAdapter.php:176]: The alignment is automaticly done by php-cs-fixer I guess..

**Pull request merged**